audbackend/core/backend/minio.py

```diff
@@ -0,0 +1,402 @@
+            secret_key=self.authentication[1],
+            secure=secure,
+            **kwargs,
+        )
+
+    @classmethod
+    def get_authentication(cls, host: str) -> typing.Tuple[str, str]:
+        """Access and secret tokens for given host.
+
+        Returns a authentication for MinIO server
+        as tuple.
+
+        To get the authentication tuple,
+        the function looks first
+        for the two environment variables
+        ``MINIO_ACCESS_KEY`` and
+        ``MINIO_SECRET_KEY``.
+        Otherwise,
+        it tries to extract missing values
+        from a config file,
+        see :meth:`get_config`.
+        If no config file exists
+        or if it has missing entries,
+        ``None`` is returned
+        for the missing entries.
+
+        Args:
+            host: hostname
+
+        Returns:
+            access token tuple
+
+        """
+        config = cls.get_config(host)
+        access_key = os.getenv("MINIO_ACCESS_KEY", config.get("access_key", None))
+        secret_key = os.getenv("MINIO_SECRET_KEY", config.get("secret_key", None))
+
+        return access_key, secret_key
+
+    @classmethod
+    def get_config(cls, host: str) -> typing.Dict:
+        """Configuration of MinIO server.
+
+        The default path of the config file is
+        :file:`~/.config/audbackend/minio.cfg`.
+        It can be overwritten with the environment variable
+        ``MINIO_CONFIG_FILE``.
+
+        If no config file can be found,
+        or no entry for the requested host,
+        an empty dictionary is returned.
+
+        The config file
+        expects one section per host,
+        e.g.
+
+        .. code-block:: ini
+
+            [play.min.io]
+            access_key = "Q3AM3UQ867SPQQA43P2F"
+            secret_key = "zuf+tfteSlswRu7BJ86wekitnifILbZam1KYY3TG"
+
+        Args:
+            host: hostname
+
+        Returns:
+            config entries as dictionary
+
+        """
+        config_file = os.getenv("MINIO_CONFIG_FILE", "~/.config/audbackend/minio.cfg")
+        config_file = audeer.path(config_file)
+
+        if os.path.exists(config_file):
+            config = configparser.ConfigParser(allow_no_value=True)
+            config.read(config_file)
+            try:
+                config = dict(config.items(host))
+            except configparser.NoSectionError:
+                config = {}
+        else:
+            config = {}
+
+        return config
+
+    def _checksum(
+        self,
+        path: str,
+    ) -> str:
+        r"""MD5 checksum of file on backend."""
+        path = self.path(path)
+        checksum = self._client.stat_object(self.repository, path).etag
+        return checksum
+
+    def _close(
+        self,
+    ):
+        r"""Close connection to repository.
+
+        An error should be raised,
+        if the connection to the backend
+        cannot be closed.
+
+        """
+        # At the moment, this is automatically handled.
```

Comment:
Automatically handled means by the minio client lib. Would it make sense to make this explicit?

Reply:
I removed `_close()` and added `close()` with an updated docstring instead:

![image](https://github.com/user-attachments/assets/e07d1ae0-bdfe-4fc3-bc37-52ef07eb1f61)

Reply:
In principle, we can also take care about opening and closing by providing an `http_client` object to MinIO, but as it will be very complicated to find out the best settings in that case, I would stay for now with the default and just state that closing is automatically handled by MinIO.